test/test_pooling.py

```diff
@@ -411,6 +412,48 @@ def find_one():
         # maxConnecting = unbounded: 30+ connections in ~0.140+ seconds
         print(len(pool.conns))
 
+    def test_csot_timeout_message(self):
+        with self.assertRaises(Exception) as error:
+            with timeout(0.5):
+                self.client.db.t.find_one({"$where": delay(2)})
```

Comment:
This is failing because the query ($where) will only be evaluated on non-empty collections. On missing/empty collections the server returns an empty result without executing the query. Same for test_socket_timeout_message below.